packages/forms/src/widgets/TogglePasswordWidget/TogglePasswordWidget.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import Password from 'react-jsonschema-form/lib/components/widgets/PasswordWidget';
+import classNames from 'classnames';
+import { renderActions } from '../../Form';
+import theme from './TogglePasswordWidget.scss';
+
+export default class TogglePasswordWidget extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.onToggle = this.onToggle.bind(this);
+		this.state = {
+			type: props.type?props.type:'password',
```

Comment:
state should not comes from props.
I would prefer this.state = { type: 'password' }
by default this seems far better